larpandoracontent/LArControlFlow/MasterAlgorithm.h

```diff
@@ -409,12 +422,30 @@ class SliceIdBaseTool : public pandora::AlgorithmTool
      *  @param  crSliceHypotheses the parent pfos representing the cosmic-ray muon outcome for each slice
      *  @param  sliceNuPfos to receive the list of selected pfos
      */
-    virtual void SelectOutputPfos(const pandora::Algorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses,
-        pandora::PfoList &selectedPfos) = 0;
+  virtual void SelectOutputPfos(const pandora::Algorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses,
+				pandora::PfoList &selectedPfos, const PfoToFloatMap &pfoToProbabilityMap, const SliceVector &sliceVector) = 0;
+};
+//-------------------------------------------------------------------------------------------------------------------------------------------
+
+/**
+*  @brief  TrackDirectionBaseTool class
+*/
+class TrackDirectionBaseTool : public pandora::AlgorithmTool
+{
+public:
+   /**
+    *  @brief  Find direction of a track
+           *
+    *  @param  parentCosmicRayPfos the list of parent cosmic-ray pfos
+    *  @param  ambiguousPfos to receive the list of ambiguous pfos 
+    *  @param  pAlgorithm the address of this master algorithm
+    */
+
+   virtual void  FindDirections(const pandora::ParticleFlowObject *const pPfo, float &downProbability, const MasterAlgorithm *const pAlgorithm) = 0;
```

Comment:
Seems to have to be, or this error occurs: 
![Screenshot from 2021-04-08 10-47-32](https://user-images.githubusercontent.com/50299856/114006474-5f660e80-9858-11eb-8ee8-e7de181e57eb.png)

Reply:
Hi Alex, If you make this change, it would need to be farmed out to all derived classes too, otherwise you'll get the error message above (the derived classes need to implement the exact function marked as pure virtual in the base class). What happens if you change this to a Pandora algorithm in the base class, and all derived classes too? Thanks.

Reply:
Hi, yes fixed. Thank you